File: lintcode/sqrt.py
class Solution:
    """
    @param x: An integer
    @return: The sqrt of x
    """
    def sqrt(self, x):
        n, nn = 1, 3
        while abs(n - nn) > 1:
            n, nn = (n + x/n) / 2, n
        return n if n * n <= x else n - 1

if __name__ == "__main__":
    solution = Solution()
    print solution.sqrt(24)

File: leetcode/sqrt.js
/* Implement int sqrt(int x).  Compute and return the square root of x.
 * Time Complexity: O(lg lg n)
 * Space Complexity: O(1)
 */

/**
 * @param {number} x
 * @return {number}
 */
var mySqrt = function(x) {
    for (var n = 1, nn = 3; Math.abs(n - nn) > 1; ) {
        nn = n
        n = Math.floor((n + Math.floor(x / n)) / 2)
    }
    return n * n <= x ? n : n - 1
};
